source/base/subscriptor.cc

```diff
@@ -185,15 +188,15 @@ Subscriptor::unsubscribe(const char *id) const
 
   --counter;
 
-  // This feature is disabled when we compile with threads: see the
-  // documentation of this class.
-#  ifndef DEAL_II_WITH_THREADS
+#ifdef DEAL_II_WITH_THREADS
+  std::lock_guard<std::mutex> lock(mutex);
+#endif
```

Comment:
I am not sure whether we actually want to serialize with a lock_guard here. Even though this is only relevant for the debug mode - I fear that a mutex here might prevent a whole class of race conditions showing up in debug mode and only in release mode. Making debugging actually harder.

Reply:
I agree with @tamiko here: we should have the same behavior in both modes.